src/crimson/osd/pg_backend.cc

```diff
@@ -784,27 +784,39 @@ PGBackend::rollback_iertr::future<> PGBackend::rollback(
                   __func__, os.oi.soid ,snapid);
   hobject_t target_coid = os.oi.soid;
   target_coid.snap = snapid;
-  return obc_loader.with_clone_obc_only<RWState::RWREAD>(
+  return obc_loader.with_clone_obc_only<RWState::RWWRITE>(
     head, target_coid,
     [this, &os, &txn, &delta_stats, &osd_op_params]
-    (auto clone_obc) {
+    (auto resolved_obc) {
+    if (resolved_obc->obs.oi.soid.is_head()) {
+      // no-op: The resolved oid returned the head object
+      logger().debug("PGBackend::rollback: loaded head_obc: {}"
+                     " do nothing",
+                     resolved_obc->obs.oi.soid);
+      return rollback_iertr::now();
+    }
+    /*
+     * The case where a CEPH_OSD_OP_ROLLBACK is not the only OSDOp
+     * included in the rados operation is *not* supported as there aren't
+     * any clients that make use of this of this kind of transaction.
+     */
```

Comment:
This isn't specific enough.  Instead:

```
/* TODO <link to tracker ticket> This implementation will not behave correctly
 * for a rados operation consisting of a mutation followed by a rollback to a
 * snapshot since the last mutation of the object.  The correct behavior would be
 * for the rollback to undo the mutation earlier in the operation by resolving
 * to the clone created at the start of the operation (see <insert method here>).
 * Instead, it will select HEAD leaving that mutation intact since the SnapSet won't
 * yet contain that clone.  This behavior exists in classic as well.
 */
```

Please create the above tracker ticket with the text of a unit test that reproduces it -- presumably on both classic and crimson.  Of course, if you can't actually reproduce it then it may not be a bug after all and we can discuss it.